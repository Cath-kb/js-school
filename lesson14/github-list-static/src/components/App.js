import React from 'react';
import UserList from './UserList';

class App extends React.Component {
  render() {
    const {users} = this.props;
    const title = 'Github User List Static';

    return (
      <div className="container my-5">
        <header className="header">
          <h1 className="title text-center mb-4">{title}</h1>
        </header>
        <UserList users={users} />
      </div>
    );
  }
}

export default App;
