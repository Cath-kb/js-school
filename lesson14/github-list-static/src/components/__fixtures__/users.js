const users = [
  {
    id: 1,
    avatar: {
      url: "https://avatars0.githubusercontent.com/u/18?v=4",
    },
    person: {
      login: "wayneeseguin",
      url: "https://api.github.com/users/wayneeseguin",
      isAdmin: false,
    }
  },
  {
    id: 2,
    avatar: {
      url: "https://avatars0.githubusercontent.com/u/19?v=4",
    },
    person: {
      login: "brynary",
      url: "https://api.github.com/users/brynary",
      isAdmin: true,
    }
  }
];

export default users;