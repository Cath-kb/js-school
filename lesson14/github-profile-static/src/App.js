import React from 'react';
import Header from './Layout/Header';
import Content from './Layout/Content';
import User from './User';

class App extends React.Component {
  render() {
    const {user} = this.props;
    const title = 'Github User Profile Static';

    return (
      <div className="container my-5">
        <Header title={title} />
        <Content>
          <User user={user} />
        </Content>
      </div>
    );
  }
}

export default App;
