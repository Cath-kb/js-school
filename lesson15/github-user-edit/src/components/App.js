import React from 'react';

import User from './User';
import Form from './User/Form';

import { getGithubUser, updateGithubUser } from '../model/api';

import { MODE_VIEW, MODE_EDIT, MODE_LOAD } from '../consts';
import { GITHUB_USERNAME } from '../configs';

class App extends React.Component {
  state = {
    mode: MODE_LOAD,
    user: null,
  };

  toggleMode = () => {
    this.setState({
      mode: this.state.mode === MODE_VIEW ? MODE_EDIT : MODE_VIEW,
    });
  };

  updateUser = (person) => {
    this.setState({
      mode: MODE_LOAD,
    });
    updateGithubUser(person)
      .then(user => this.setState({
        mode: MODE_VIEW,
        user,
      }));
  };

  isMode(mode) {
    return this.state.mode === mode;
  }

  componentDidMount() {
    getGithubUser(GITHUB_USERNAME)
      .then(user => this.setState({
        mode: MODE_VIEW,
        user,
      }));
  }

  render() {
    const { user } = this.state;

    const title = `Github User Profile Fetching and Updating ${this.isMode(MODE_EDIT) ? '(Edit)': ''}`;

    return (
      <div className="container my-5">
        <header className="header">
          <h1 className="title text-center mb-4">{title}</h1>
        </header>
        {this.isMode(MODE_VIEW) && <User user={user} onClick={this.toggleMode} />}
        {this.isMode(MODE_EDIT) && <Form initialUser={user} onReset={this.toggleMode} onSubmit={this.updateUser} />}
        {this.isMode(MODE_LOAD) && <h2 className="my-5 text-center">Loading&hellip;</h2>}
      </div>
    );
  }
}

export default App;
