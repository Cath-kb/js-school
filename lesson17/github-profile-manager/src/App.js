import React from 'react';
import { Switch, Route, Redirect } from 'react-router';

import Profile from "./components/Profile";
import EditForm from "./components/EditForm";

import { getGithubUser } from './model/api';

class App extends React.Component {
  state = {
    isLoading: true,
    user: null,
  };

  getUser(user) {
    getGithubUser(user)
      .then(user => this.setState({
        isLoading: false,
        user,
      }));
  }

  handleSuccess = user => {
    this.setState({
      user,
    });
  };

  componentDidMount() {
    const { user_login } = this.props.match.params;
    this.getUser(user_login);
  }

  componentDidUpdate(prevProps) {
    const { user_login } = this.props.match.params;

    if (user_login !== prevProps.match.params.user_login) {
      this.getUser(user_login);
    }
  }

  render() {
    const title = 'Github Profile Manager React App';

    const { user_login } = this.props.match.params;
    const { user, isLoading } = this.state;

    return (
      <div className="container my-5">
        <header className="header">
          <h1 className="title text-center mb-4">{title}</h1>
        </header>
        {
          isLoading ? <h2 className="my-5 text-center">Loading&hellip;</h2> :
          <Switch>
            <Route path="/:user_login/edit" render={props => <EditForm initialUser={user} {...props} onSuccess={this.handleSuccess} />} />
            <Route path="/:user_login" render={props => <Profile user={user} {...props} />} />
            <Route render={() => <Redirect to={`/${user_login}`} />} />
          </Switch>
        }
      </div>
    );
  }
}

export default App;
