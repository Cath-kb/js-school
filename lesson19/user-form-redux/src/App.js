import React, { Component } from 'react';
import UserForm from './UserForm';
import UserInfo from './UserInfo';

// DO NOT Modify this file

class App extends Component {
  render() {
    return (
      <div className="container">
        <header className="header">
          <h1>User Form</h1>
        </header>
        <UserForm />
        <UserInfo />
      </div>
    );
  }
}

export default App;
