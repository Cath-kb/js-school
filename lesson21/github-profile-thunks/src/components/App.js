import React from 'react';
import { connect } from 'react-redux';

import { load } from '../store';

import { Header, Content, Loading, Error } from './Layout';
import User from './User';

class App extends React.Component {
  componentDidMount() {
    const { user_login } = this.props.match.params;
    this.props.loadUser(user_login);
  }

  render() {
    const { isLoading, user, error } = this.props;
    const title = 'Github User Profile Redux Thunks';

    return (
      <div className="container my-5">
        <Header title={title} />
        {isLoading ?
          <Loading /> :
          user ?
          <Content>
            <User user={user} />
          </Content> :
          error ?
          <Error error={error} /> :
          null
        }
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isLoading: state.isLoading,
  user: state.user,
  error: state.error,
});

const mapDispatchToProps = dispatch => ({
  loadUser: user_login => dispatch(load(user_login)),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
