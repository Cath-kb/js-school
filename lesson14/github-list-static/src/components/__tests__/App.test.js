import React from 'react';
import ReactDOM from 'react-dom';
import renderer from "react-test-renderer";
import App from '../App';
import users from '../__fixtures__/users';

const app = {
  title: 'Test title',
  users,
};

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App users={app.users} title={app.title}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should be rendered without errors', () => {
    const html = renderer.create(<App users={app.users} title={app.title} />).toJSON();
    expect(html).toMatchSnapshot();
  });
});
