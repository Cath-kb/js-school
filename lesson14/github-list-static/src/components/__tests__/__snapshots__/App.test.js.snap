// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be rendered without errors 1`] = `
<div
  className="container my-5"
>
  <header
    className="header"
  >
    <h1
      className="title text-center mb-4"
    >
      Github User List Static
    </h1>
  </header>
  <div
    className="row my-4"
  >
    <div
      className="col-sm-2"
      data-id="user"
    >
      <img
        alt="User Avatar"
        className="w-100 img-fluid rounded mb-4 mb-sm-0"
        src="https://avatars0.githubusercontent.com/u/18?v=4"
      />
    </div>
    <div
      className="col-sm-10"
    >
      <h2
        className="d-flex align-items-center"
      >
        wayneeseguin
      </h2>
      <h3>
        <a
          href="https://api.github.com/users/wayneeseguin"
          target="_blank"
          title="Go to user profile"
        >
          https://api.github.com/users/wayneeseguin
        </a>
      </h3>
    </div>
  </div>
  <div
    className="row my-4"
  >
    <div
      className="col-sm-2"
      data-id="user"
    >
      <img
        alt="User Avatar"
        className="w-100 img-fluid rounded mb-4 mb-sm-0"
        src="https://avatars0.githubusercontent.com/u/19?v=4"
      />
    </div>
    <div
      className="col-sm-10"
    >
      <h2
        className="d-flex align-items-center"
      >
        brynary
        <span
          className="ml-4 badge badge-pill badge-dark"
        >
          admin
        </span>
      </h2>
      <h3>
        <a
          href="https://api.github.com/users/brynary"
          target="_blank"
          title="Go to user profile"
        >
          https://api.github.com/users/brynary
        </a>
      </h3>
    </div>
  </div>
</div>
`;
